server/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java

```diff
@@ -249,7 +237,7 @@ public UnicastZenPing(Settings settings, ThreadPool threadPool, TransportService
                 logger.warn("timed out after [{}] resolving host [{}]", resolveTimeout, hostname);
             }
         }
-        return discoveryNodes;
+        return transportAddresses;
```

Comment:
I'd expect this to return an ummutable list of transport addresses, just in case the list is passed around...?

Reply:
good point. I've pushed d0db5e5fa2e